docker/emptydrops/emptyDropsWrapper/emptyDropsWrapper.R

```diff
@@ -0,0 +1,211 @@
+		action='store_true',
+		help='transpose the input matrix before processing',
+		default=FALSE),
+    make_option(c('--emptydrops-lower'),
+                default=100, ## as per emptyDrops package
+                help='emptydrops lower parameter',
+                dest='ed_lower'),
+    make_option(c('--emptydrops-niters'),
+                default=10000, ## as per emptyDrops package
+                help='emptydrops niters paramters',
+                dest='ed_niters'),
+    make_option(c('--emptydrops-test-ambient'),
+                dest='ed_test_ambient',
+                default=FALSE, ## as per emptyDrops package
+                action='store_true',
+                help='Empty drops parameter test.ambient'),
+    make_option(c('--emptydrops-ignore'),
+                default=NULL, ## as per emptyDrops package
+                help='emptyDrops ignore parameter',
+                dest='ed_ignore'),
+    make_option(c('--emptydrops-alpha'),
+                default=NULL, ## as per emptyDrops package
+                help='emptyDrops alpha parameter',
+                dest='ed_alpha')
+    ## TODO: Expose parallel functionality
+)
+
+## Parse the arguments
+opt_parser <- OptionParser(option_list=option_list);
+opt <- parse_args(opt_parser);
+
+## Check the parsed arguments
+if(is.null(opt$input_rds))  errorExit("Input RDS is not specified\n");
+if(is.null(opt$output_csv)) errorExit("Output CSV is not specified\n");
+if(!file.exists(opt$input_rds)) errorExit("Input RDS doesn't exist!\n");
+if(file.exists(opt$output_csv)) errorExit("Output CSV file exists!\n");
+
+## Load the required libraries here
+## NOTE: We do this after parsing arguments so that --help returns immediatedly
+pkgs <- c('DropletUtils','Matrix')
+missing_pkgs <- pkgs[!pkgs %in% installed.packages()]
+if(length(missing_pkgs) > 0) {
+    cat('Error the following package(s) are missing:', paste0(missing_pkgs, collapse=', '),'\n')
+    quit(save="no",status=1)
+}
+
+## Load the libs, suppressing warnings and other messages
+suppressWarnings(suppressPackageStartupMessages(library('DropletUtils',quietly=TRUE)))
+suppressWarnings(suppressPackageStartupMessages(library('Matrix',quietly=TRUE)))
+
+## Set script parameters, this need to be done after package loading
+## as SerialParam() is not otherwise availbale
+inputRDS <- opt$input_rds
+output_csv <- opt$output_csv
+FDRcutoff <- opt$fdr_cutoff
+verbose <- opt$verbose
+ed_param_lower <- opt$verbose
+ed_param_niters <- opt$ed_niters
+ed_param_test.ambient <- opt$ed_test_ambient
+ed_param_ignore <- opt$ed_ignore
+ed_param_alpha <- opt$ed_alpha
+ed_param_BPPARAM <- SerialParam() ## TODO: Allow serial or parallele, with ncore specification
+
+## Read the input file
+inputMatrix <- readRDS(inputRDS)
+
+## Check the input matrix
+## Check class
+if(!class(inputMatrix) %in% c( 'dgCMatrix','dgRMatrix' )) {
+    cat(paste0('Error: input matrix is not of class dgCMatrix or dgRMatrix. It is of class:',class(inputMatrix)),file=stderr())
```

Comment:
nit: It'd be better to print messages even there are no errors occurred here. (cat in both if and else branch). Sometimes not only the stderrs are useful but the stdouts are also valuable to be elaborate. Same comments for all of the checkers.

E.g.:
```
if error:
    cat('error!')
cat('succeeded!')
```

Reply:
I don't want to make the output noisy by default. I'll make it print these messages if in verbose mode.

Reply:
Makes sense